File: univalued-binary-tree/univalued-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isUnivalTree(self, root: Optional[TreeNode]) -> bool:
        dq = collections.deque([root])
        while dq:
            node = dq.popleft()
            if (node.val) != root.val:
                return False
            dq.extend([n for n in (node.left,node.right) if n])
            
        return True

File: univalued-binary-tree/univalued-binary-tree.swift
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public var val: Int
 *     public var left: TreeNode?
 *     public var right: TreeNode?
 *     public init() { self.val = 0; self.left = nil; self.right = nil; }
 *     public init(_ val: Int) { self.val = val; self.left = nil; self.right = nil; }
 *     public init(_ val: Int, _ left: TreeNode?, _ right: TreeNode?) {
 *         self.val = val
 *         self.left = left
 *         self.right = right
 *     }
 * }
 */
class Solution {
    func isUnivalTree(_ root: TreeNode?) -> Bool {
        var val = root!.val
        return helper(root,val)
    }
    
    func helper(_ root: TreeNode?, _ val:Int) -> Bool {
        if let node = root {
            return node.val != val ? false : true && helper(node.left,val)
 && helper(node.right, val)        }
        return true
    }
}